mypy/subtypes.py

```diff
@@ -297,16 +295,21 @@ def visit_instance(self, left: Instance) -> bool:
                     return True
                 if isinstance(item, Instance):
                     return is_named_instance(item, 'builtins.object')
-        if isinstance(right, LiteralType) and left.last_known_value is not None:
-            return self._is_subtype(left.last_known_value, right)
         if isinstance(right, CallableType):
             # Special case: Instance can be a subtype of Callable.
             call = find_member('__call__', left, left, is_operator=True)
             if call:
                 return self._is_subtype(call, right)
             return False
-        else:
-            return False
+        # Special cases
+        if isinstance(right, LiteralType) and left.last_known_value is not None:
+            return self._is_subtype(left.last_known_value, right)
+        right_fallback = mypy.typeops.try_getting_instance_fallback(right)
+        if (right_fallback is not None
+                and ((isinstance(right, TupleType) and right_fallback.type.is_enum)
+                     or isinstance(right, TypedDictType))):
```

Comment:
IIRC the fallback of a TypedDict type is actually not a "real" type, but a sort of implementation artifact. I believe that the actual bug is that we leak the fallback as an instance type. In particular, it shouldn't be possible to bind a TypedDict type to `Type[T]`, since a TypedDict type is not a real class, and doesn't support `isinstance`, among other things. I suspect that is where things go wrong, and this change would be just papering over the root cause. I'm not 100% sure what is going on here, though.

Reply:
> the actual bug is that we leak the fallback as an instance type

Yes you're right. I am confused by `Type[T]` before and I just saw https://github.com/python/mypy/issues/9773 and other related discussions. If my understanding is correct, these issues should be fiexd by two steps: 1. throw errows when binding a TypedDict to `Type[T]`. 2. implement `TypeForm[t]` to support the original design pattern described in those issues.

Reply:
oops, I still need some extra investigation to catch up disscussions on `Type[T]` and `TypeForm`. It'd be great if you could provide some suggestions on implementation since it's a quite common issue.